lib/rustler_precompiled.ex

```diff
@@ -534,6 +593,38 @@ defmodule RustlerPrecompiled do
     end
   end
 
+  defp variants_for_metadata(variants) do
+    Map.new(variants, fn {target, values} -> {target, Keyword.keys(values)} end)
+  end
+
+  # Extract the target without the nif-NIF-VERSION part
+  defp target_triple_from_nif_target(nif_target) do
+    [_, _, triple] = String.split(nif_target, "-", parts: 3)
```

Comment:
The idea is to take the target from a string like this: `nif-2.16-x86_64-unknown-linux-gnu`, so this would not work.
I forced the pattern to match the `nif` at the beginning, so we won't be using this function with strings that are not in that format. WDYT?

Reply:
Oh, I see. So the previous version worked for me too. Your call!

Reply:
The current version is clearer though, so I would keep it.

Reply:
I will keep this way (matching the "nif") :)